tests/expr_and_series/unique_test.py

```diff
@@ -1,20 +1,49 @@
 from __future__ import annotations
 
-import narwhals.stable.v1 as nw
+from contextlib import nullcontext as does_not_raise
+
+import pytest
+
+import narwhals as nw
+from narwhals.exceptions import LengthChangingExprError
+from tests.utils import Constructor
 from tests.utils import ConstructorEager
 from tests.utils import assert_equal_data
 
 data = {"a": [1, 1, 2]}
 data_str = {"a": ["x", "x", "y"]}
 
 
-def test_unique_expr(constructor_eager: ConstructorEager) -> None:
-    df = nw.from_native(constructor_eager(data))
-    result = df.select(nw.col("a").unique())
-    expected = {"a": [1, 2]}
+def test_unique_expr(constructor: Constructor) -> None:
+    df = nw.from_native(constructor(data))
+    context = (
+        pytest.raises(LengthChangingExprError)
+        if isinstance(df, nw.LazyFrame)
+        else does_not_raise()
+    )
+    with context:
+        result = df.select(nw.col("a").unique())
+        expected = {"a": [1, 2]}
+        assert_equal_data(result, expected)
+
+
+def test_unique_expr_agg(
+    constructor: Constructor, request: pytest.FixtureRequest
+) -> None:
+    if any(x in str(constructor) for x in ("duckdb", "pyspark")):
+        request.applymarker(pytest.mark.xfail)
+    df = nw.from_native(constructor(data))
+    result = df.select(nw.col("a").unique().sum())
+    expected = {"a": [3]}
     assert_equal_data(result, expected)
 
 
```

Comment:
Can we maybe add a test with a length changing and an aggregation together? This should not raise I think.

e.g.

```suggestion


def test_unique_expr_and_agg(
    constructor: Constructor, request: pytest.FixtureRequest
) -> None:
    if any(x in str(constructor) for x in ("duckdb", "pyspark")):
        request.applymarker(pytest.mark.xfail)
    df = nw.from_native(constructor(data))
    result = df.select(nw.col("a").unique().alias("unique"), nw.col("a").sum().alias("sum"))
    expected = {"unique": [1, 2], "sum": [4]}
    
    assert_equal_data(result, expected)


```

WDYT?

Reply:
i'm confused, sorry - `expected` has keys of different lengths?

Reply:
Whops, my bad! Just updated the values. It's a length changing and an aggregated that then get broadcasted in the same select context.

Reply:
thanks! I think this is the kind of operation that sql / pyspark / wouldn't support, and I'm not sure we need to, it could just as well be done with `df.select('a', nw.col('a').sum().alias('sum')).unique('a')`

trying this out for example raises:
```python
In [14]: rel = duckdb.sql('select * from values (1, 4), (1, 2), (2, 3), (2, 4) df(a, b)')

In [15]: duckdb.sql("""
    ...: select unnest(list(distinct a)), sum(b) over () from rel
    ...: """)
---------------------------------------------------------------------------
BinderException                           Traceback (most recent call last)
Cell In[15], line 1
----> 1 duckdb.sql("""
      2 select unnest(list(distinct a)), sum(b) over () from rel
      3 """)

BinderException: Binder Error: column "b" must appear in the GROUP BY clause or must be part of an aggregate function.
Either add it to the GROUP BY list, or use "ANY_VALUE(b)" if the exact value of "b" is not important.
```

It would still be backwards-compatible if we later allowed it - but it's easier to start with a simpler and predictable rule, and if necessary relax later

Reply:
Cool! Resolving then!